Some improvements

adam-waldenberg opened the pull request:
These changes slightly improve the highlighting while removing unnecessary code. It also adds better support for different versions of PyFlakes (it should probably support them all).

adam-waldenberg committed:
Remove seemingly unneeded BlackHole class

After some initial tests it seems this class is not needed at all? Is
there maybe a difference in behavior between Python 2 and 3?

adam-waldenberg committed:
Change messages.RedfinedFunction to messages.Redefined

The other one seems to have been removed in newer PyFlakes versions. The
whole thing needs to be done differently, as the PyFlakes messages module
seems to have a tendency to change a lot.

adam-waldenberg committed:
Implement a more clear distinction between warnings and errors

Errors include faults that will cause a runtime error of some kind when
the Python script is executed. Warnings cover simpler mistakes such as
unused variables and unused imports.

This change also has the positive side effect that it makes the plugin
work better with most (if not all) PyFlakes versions, as it no longer
directly references attributes inside the messages module.

adam-waldenberg committed:
Move syntax error exception catching out of the check() method

This removes the need for the PyLocation and PySyntaxError classes.

adam-waldenberg committed:
Implement find_keyword() method

This moves the keyword scanning out of the show_errors() method and
makes it slightly more readable.

adam-waldenberg committed:
Always mark starting at the content instead of the beginning of lines

This skips white space characters when marking syntax errors from the
parser. Introduced a get_line_interval() method in order to make the
solution slightly cleaner.

aniav commented:
Personally I don't like this code style but I think the checks should be done by @patrys, he wrote most of this code and I only did some cosmetic fixes afterwards. I am also unable to test it because of the OS I am using.

adam-waldenberg commented:
@aniav 

To what are you referring? I tried to follow the same code style as previously used.

adam-waldenberg committed:
Use single quotes in PYFLAKES_MESSAGES_WARNING

This better follows the previous convention in the code.

adam-waldenberg committed:
Re-implement asterisk (*) highlighting on ImportStarUsage warning

This was previously removed in the commit ad5dbe7a.

adam-waldenberg commented:
Did a rebase onto upstream master.

adam-waldenberg committed:
recheck() is now do_recheck() to clarify it's called via a signal

adam-waldenberg committed:
Show a description tooltip when hovering over a highlight

This description is directly taken from the pylint message or from the
syntax error exception reported from the AST parser.

The tooltip handler supports both keyboard and mouse queries.
